### PR TITLE
Add tests for extra fields for `affiliation`

### PR DESCRIPTION
**Pull request by potsui**
@caesar2164 @caseylitton @stvstnfrd PR for added tests to CME registration form, `clean_affiliation`

When `affiliation` is selected to be "Stanford University,
`stanford_department` and `sunet_id` is required. Function
`clean_affiliation` in `forms.py` checks this. Previously fixed bug
regarding frozen registration screen on invalid inputs (PR #650) and
discovered missing unit tests. Added tests check that form is invalid
given `sunet_id` that is missing or not 3-8 chars long, or
`stanford_department` that is missing or invalid.

CC: @jspayd @jlikhuva

### CODE REVIEW

**Comment by caesar2164** [comment]
@potsui, could you also add a test for an empty "postal code"? (my recent PR also fixed that one)

**Comment by caesar2164** [comment]
awesome, thanks @potsui !

File: openedx/stanford/djangoapps/register_cme/tests/test_form.py [hunk 1]
```diff
@@ -151,3 +151,31 @@ def test_export_countries(self, country):
...
+        data = self.data
+        data['affiliation'] = 'Stanford University'
+        data['stanford_department'] = 'Medicine'
+        data['sunet_id'] = sunet_id
+        form = ExtraInfoForm(data)
+        is_valid = form.is_valid()
+        self.assertFalse(is_valid)
+
+    @data(None, 'invalid_dept')
+    def test_clean_affiliation_stanford_department(self, stanford_department):
```
Comment (stvstnfrd) [inline openedx/stanford/djangoapps/register_cme/tests/test_form.py]: Let's also rename this: `test_affiliation_stanford_department`

File: openedx/stanford/djangoapps/register_cme/tests/test_form.py [hunk 2]
```diff
@@ -151,3 +151,31 @@ def test_export_countries(self, country):
...
+    def test_clean_affiliation_sunet_id(self, sunet_id):
+        data = self.data
+        data['affiliation'] = 'Stanford University'
+        data['stanford_department'] = 'Medicine'
+        data['sunet_id'] = sunet_id
+        form = ExtraInfoForm(data)
+        is_valid = form.is_valid()
+        self.assertFalse(is_valid)
+
+    @data(None, 'invalid_dept')
```
Comment (stvstnfrd) [inline openedx/stanford/djangoapps/register_cme/tests/test_form.py]: It's generally beneficial to avoid abbreviations (clarity over brevity for maintainability),
so we can just call this `invalid_department`.

**Comment by potsui** [comment]
@stvstnfrd @caesar2164 fixed requested changes

File: openedx/stanford/djangoapps/register_cme/tests/test_form.py [hunk 3]
```diff
@@ -151,3 +151,31 @@ def test_export_countries(self, country):
...
+    def test_affiliation_stanford_department(self, stanford_department):
+        data = self.data
+        data['affiliation'] = 'Stanford University'
+        data['stanford_department'] = stanford_department
+        data['sunet_id'] = 'sunet_id'
+        form = ExtraInfoForm(data)
+        is_valid = form.is_valid()
+        self.assertFalse(is_valid)
+
+    @data('1')
```
Comment (stvstnfrd) [inline openedx/stanford/djangoapps/register_cme/tests/test_form.py]: Hrm, we probably shouldn't use DDT if we aren't providing more than one value.

We could either:
- remove the `@data` decorator and just hard code the postal_code to the value
- add an additional value and also test for too long of an entry (I forget what the max value is)
  - if we do that, we'll want to modify the method name.

Comment (potsui) [inline openedx/stanford/djangoapps/register_cme/tests/test_form.py]: It seems like `clean_postal_code` in the form only enforces a minimum length, so the former option seems fine?

**Comment by potsui** [comment]
@stvstnfrd @caesar2164 2nd round review fixed, I also added empty string test for `sunet_id` since it's along the same lines as `stanford_department`

**Comment by stvstnfrd** [comment]
@potsui This looks good; let's squash then merge.

**Comment by potsui** [comment]
@stvstnfrd Squashed.

**Comment by caesar2164** [comment]
uh oh, looks like you squashed a commit incorrectly?

**Comment by caesar2164** [comment]
test this please
